### PR TITLE
Add balance recipe

### PR DESCRIPTION
**Pull request by sergiruiztrepat**
### Brief summary of what the package does

Balance chemical equations.

### Direct link to the package repository

https://github.com/sergiruiztrepat/chembalance

### Your association with the package

Author.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

**Comment by purcell** [comment]
Just a quick comment (haven't reviewed the code yet): I'd suggest a more unique name, e.g. `chembalance`, to avoid confusion w.r.t. balancing things in general, e.g. the built-in `balance-windows` function. It's generally good to avoid using prefixes that already exist.

**Comment by riscy** [comment]
Agreed -- the first thing that leapt to my mind was balancing parentheses. :)

Here's a very quick pass over the checklist in the meantime:

### balance.el ###

byte-compile (using Emacs 27.1):
```
balance.el:311:31:Warning: assignment to free variable `val'
balance.el:311:31:Warning: reference to free variable `val'
In balance-coeficients:
balance.el:428:13:Warning: assignment to free variable `compound'
balance.el:432:35:Warning: reference to free variable `compound'
In end of data:
balance.el:628:1:Warning: the function `lcm' is not known to be defined.
```

checkdoc (using version 0.6.2):
- No issues!

package-lint (using version 20210425.3):
- No issues!

Other [lints](https://github.com/riscy/melpazoid):
- balance.el#L99: Optionally use `unless ...` instead of `when (not ...)` (this is minor, and up to you; there are a few instances of this I see)

Also rather than `... Balance chemical equations - * -lexical-binding: t:- * -` your first line should look like `... Balance chemical equations -*- lexical-binding: t -*-`, otherwise you'll corrupt your package summary.

**Comment by sergiruiztrepat** [comment]
Thanks to both,
I've changed name to "chembalance" but doing this changes I think I messed something and github fails checks.
Now package is chembalance, and also the repo is https://github.com/sergiruiztrepat/chembalance.
Other little changes from @riscy are done.

File: recipes/chembalance [hunk 1]
```diff
@@ -0,0 +1 @@
+(balance :fetcher github :repo "sergiruiztrepat/chembalance")
```
Comment (purcell) [inline recipes/chembalance:1]:
```suggestion
(chembalance :fetcher github :repo "sergiruiztrepat/chembalance")
```

**Comment by purcell** [comment]
Looking good, will merge now. Sent you a small fix in https://github.com/sergiruiztrepat/chembalance/pull/2.

**Pull request merged**